Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM node:14.15-alpine as build
+
+## Variables
+# User and ID
+ARG APP=brasil-api 
+ARG ID=1001
+
+# Default application port "from next framework"
+ARG PORT=3000
+
+
+## Commands
+# Add User
+RUN     adduser --uid ${ID} -D -h /home/${APP} ${APP} node 
+# Set User
+USER ${APP}
+#Copy files
+COPY --chown=${ID}:node . /home/${APP}
+# Set work directory
+WORKDIR /home/${APP}
```

Comment:
Então, não é comum fazer esse tipo de coisa numa imagem de produção.
Normalmente, o serviço roda como `root`, não se cria usuários, o workdir não fica na pasta do usuário e por aí vai.
Então você pode ignorar esse trecho de código e fazer a cópia e forçar o `WORKDIR` para `/app` (mais comum), `/code`, `/brapi` e por aí vai.

```Dockerfile
COPY . /app/
WORKDIR /app
# ou
WORKDIR /app
COPY . .
```

Reply:
Então, dois pontos.:

- Workdir.:

Sim, movi a para /brasil-api , assim fica padrão a foder.

- USER.:

Correr qualquer coisa que como root é muito perigoso, tenho que partir do principio de privilégios minimos para uma aplicação. Somente as permissões necessárias para executar a tarefa.

Em uma execução local, é pouco problematico. Porem caso isso vá para um k8s ou uma cloud conteiner.

> Se for root dentro de um pod de k8s, e alguem chegar ao servidor. Ele consegue navegar para as outras aplicações do pod, ou dependendo, instalar programas para correr.

> Tambem podemos ter problemas com a aplicação e sua execução ao correr com .: 
```
apiVersion: v1
kind: Pod
metadata:
  name: brasil-api-host
spec:
  securityContext:
    runAsUser: 1000
    fsGroup: 1000
  containers:
  - name: brasil-api
    image: supergiantkir/k8s-liveliness
    securityContext:
      allowPrivilegeEscalation: false
```
Com essa configuração se o user padrão da imagem for o *0* *root*, em um override de permissão o novo user não consegue rodar nenhum ficheiro.

> Tambem temos o uso de policies.: 
```
 .spec.securityContext.runAsNonRoot